src/Form/Formbot.js

```diff
@@ -137,8 +149,16 @@ export default class Formbot extends React.Component {
       let errorMsg;
 
       try {
-        if (fromSchema) {
-          validation.validateSync(fieldValue, { values: this.state.values });
+        if (hasSchema) {
+          if (typeof validation.validate === 'function' && isAsyncValidation) {
+            validation.validate(fieldValue, validationOpts).catch(e => {
+              this.setErrors({ [field]: e.message }, resolve);
+            });
+
+            return;
+          }
+
+          validation.validateSync(fieldValue, validationOpts);
```

Comment:
Is it necessary to make a distinction between sync/async validations?  I'm assuming we could always call `validation.validate` with a sync or async function and have it resolve correctly.  I did a quick test in the `molekule` example server setting `async: true` in the schema and passing a non-async function to `test()`.  It still validates correctly so that is why I'm thinking that maybe this logic could be simplified.  Let me know if i'm understanding this incorrectly.

Reply:
Good observation, will fix 👍